### PR TITLE
Remove EOL 2.7

### PR DESCRIPTION
**Pull request by TimWolla**

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -59,7 +59,7 @@ ENV HAPROXY_SHA256 {{ .sha256 }}
 		# Since 5.4 is supported on haproxy, better use it now, but only for
 		# newer versions since there could be some minor incompatibilities
 		# for existing scripts: https://www.lua.org/manual/5.4/manual.html#8
-		if ([ "2.0", "2.2", "2.4", "2.6", "2.7", "2.8" ] | index(env.version)) then
```
Comment (LaurentGoderre) [inline Dockerfile.template:62]: 2.0 is also out of LTS and also EOL

**Review by LaurentGoderre — APPROVED** [review]

File: Dockerfile.template (hunk 1 above)
Comment (TimWolla) [reply Dockerfile.template:62]: 2.0 is not EOL.

Comment (LaurentGoderre) [reply Dockerfile.template:62]: i found 2 sources that said it is and one that was ambiguous. It can be done in a separate PR

Comment (TimWolla) [reply Dockerfile.template:62]: I don't know what sources you used, but the official project homepage does not list HAProxy 2.0 as EOL: https://www.haproxy.org/. And neither does yesterday's release announcement of HAProxy 2.0.35 (whereas the one for 2.7.12 did): https://www.mail-archive.com/haproxy@formilux.org/

**Review by yosifkit — APPROVED** [review]

**Comment by yosifkit** [comment]
We'll wait until after https://github.com/docker-library/official-images/pull/16555, so that we push the final release.

**Comment by tianon** [comment]
(Rebased thanks to https://github.com/docker-library/haproxy/pull/226 :sweat_smile:)

**Comment by tianon** [comment]
and now https://github.com/docker-library/official-images/pull/16555 is not just merged but fully built and published :rocket:

**Comment by tianon** [comment]
I don't think we really need to wait for CI here :smile:

**Pull request merged**